.github/workflows/backport.yml

```diff
@@ -43,6 +43,15 @@ jobs:
               name: 'backport: auto'
             })
 
+  list-target-branches:
+    runs-on: ubuntu-latest
+    outputs:
+      matrix: ${{ steps.get-branch-list.outputs.matrix }}
+    steps:
+      - uses: actions/checkout@v2
+      - id: get-branch-list
+        run: echo "::set-output name=matrix::$(yq '.target_branch_list' ./etc/target_branches.yml)"
```

Comment:
So has similar content but it's not the same, so parsing would be more complicated.

Parsing the data in `stack_schema_map.yaml` would change the `run` command to something like :


`echo "::set-output name=matrix::[$(yq eval '... comments=""|keys| del(.[length - 1])|.[] |= sub("(.0)$", "") | join(", ")' etc/stack-schema-map.yaml)]"`

- `...comments = ""` : remove comments or `#`
- `| keys ` : only give me the version numbers (e.g. 8.3.0)
- `| del(.[length - 1])` : delete the last item (e.g. 8.4)
- | `.[]` : give us an entire array to operate on
- `|= sub("(.0)$", "")` : remove the extra `.0` at the end since we have versions like `8.0.0` that won't match the branch versions of `8.0`
- `| join(", ")` : convert to a flat comma separated list


Benefit is we can parse from the same file, con is that it's complicated. Alternative options most likely will be multiline bash commands.

Reply:
I think either is fine. Due to timing of this as well as the changes to reference a single file is why I added the comment. If we add another file, we just have to adjust documentation to ensure we edit that file during prepping for the next release cycle is all.